https://github.com/lautr3k/lw.svg-parser/issues/4

jorgerobles opened the pull request:
Added silly symbol support  (only visualization, treated as a def.)

jorgerobles committed:
https://github.com/lautr3k/lw.svg-parser/issues/4

lautr3k commented:
Cool! 
Before i merge your commit, could you change
```
let element = this.parser.defs[target]
if (! element)
    element = this.parser.symbols[target];
```

by 

`let element = this.parser.defs[target] || this.parser.symbols[target];`

Thanks (btw i always use brackets, even when there is only one instruction)

jorgerobles committed:
https://github.com/lautr3k/lw.svg-parser/issues/4

jorgerobles commented:
Done!

Commit:
Skip <symbol> tag (not rendered)

> Note that a symbol element itself is not rendered. Only instances of a symbol element (i.e., a reference to a symbol by a <use> element) are rendered.

https://developer.mozilla.org/fr/docs/Web/SVG/Element/symbol

Commit:
Skip empty <defs>, <use> and <symbol> tags

jorgerobles committed:
https://github.com/lautr3k/lw.svg-parser/issues/4

jorgerobles commented:
Hmm with https://github.com/lautr3k/lw.svg-parser/pull/5/commits/bf11bcab26427baf86cc4115eb8982349fc9c45f the symbol is never rendered on `<use>`. The no rendering is correct, but then the symbol children should be appended as `use` children.

I've added array support for `use` but is surely wrong as replaces the use tag with the instance. Propose a non-standard element `<instance>` or `<g>` to wrap the symbol childnodes.

lautr3k commented:
I worked on this all the day and I think there's a big problem in the heart of the parser... I will resume the investigations quietly tomorrow...

lautr3k commented:
I have found a way to have full CSS support ;) and a better way to handle attributes, I am refactoring the entire library...

jorgerobles commented:
Cheer Up!
